.travis.yml

```diff
@@ -57,6 +66,17 @@ after_success:
             docker push welder/web-with-coverage
             docker push welder/web-e2e-tests
         fi
+  - |
+        if [ "$COMMAND" == "cockpit-test" ]; then
+            nyc report
+            nyc report --reporter=lcov | codecov
+        fi
+
+        if [ "$COMMAND" == "cockpit-test" ] && [ "$TRAVIS_BRANCH" == "master" ] && [ "$TRAVIS_PULL_REQUEST" == "false" ]; then
```

Comment:
Same comment as above. Check `TRAVIS_BRANCH` & `TRAVIS_PULL_REQUEST` first; `docker login` and then based on which command we are executing `docker tag && push` the resulting images.

Reply:
It's OK, the downside is there'll be an useless docker login for tests other than e2e and cockpit.

Reply:
yeah but simpler and cleaer code trumps that downside.